bpf/bpf_lxc.c

```diff
@@ -552,12 +562,19 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx,
 			   orig_dip, *dstID);
 	}
 
-	/* If the packet is in the establishing direction and it's destined
-	 * within the cluster, it must match policy or be dropped. If it's
-	 * bound for the host/outside, perform the CIDR policy check.
+	/* When an endpoint connects to itself via service clusterIP, we need
+	 * to skip the policy enforcement. If we didn't, the user would have to
+	 * define policy rules to allow pods to talk to themselves. We still
+	 * want to execute the conntrack logic so that replies can be correctly
+	 * matched.
 	 */
-	verdict = policy_can_egress4(ctx, &tuple, SECLABEL, *dstID,
-				     &policy_match_type, &audited);
+	if (!hairpin_flow)
+		/* If the packet is in the establishing direction and it's destined
+		* within the cluster, it must match policy or be dropped. If it's
+		* bound for the host/outside, perform the CIDR policy check.
+		*/
```

Comment:
bpf checks are picking up the whitespace nit here
```suggestion
		 * within the cluster, it must match policy or be dropped. If it's
		 * bound for the host/outside, perform the CIDR policy check.
		 */
```